components/admin_console/admin_definition.jsx

```diff
@@ -4124,3 +4125,28 @@ t('admin.field_names.postEditTimeLimit');
 t('admin.field_names.restrictCreationToDomains');
 t('admin.field_names.restrictDirectMessage');
 t('admin.field_names.teammateNameDisplay');
+
+let cache = null;
+const adminDefinitionsPlugins = {};
+
+export function registerAdminConsolePlugin(pluginId, func) {
+    adminDefinitionsPlugins[pluginId] = func;
+    cache = null;
+}
+
+export function unregisterAdminConsolePlugin(pluginId) {
+    delete adminDefinitionsPlugins[pluginId];
+    cache = null;
+}
+
+export default function getAdminDefinition() {
+    if (cache === null) {
+        let result = cloneDeep(AdminDefinition);
```

Comment:
I always deep clone the "root" admin definitions to avoid any update from the plugin functions, this way we can be sure that plugins are correctly installed and uninstalled without altering the original Admin definition.